src/main/scala/net/snowflake/spark/snowflake/ServerConnection.scala

```diff
@@ -109,19 +109,35 @@ private[snowflake] object ServerConnection {
   def getServerConnection(parameters: MergedParameters, useCache: Boolean = true)
   : ServerConnection = synchronized {
     val connectionCacheKey = new ConnectionCacheKey(parameters)
-    val enableCache = useCache && connectionCacheKey.isConnectionCacheSupported
+    // when use connection from Snowpark, always reuse cached the connection
+    // since the Snowpark connection in the staging area will be purged after get.
```

Comment:
when connection id provided, Spark connector will enforce to reuse connections, since it can't create new connection with the given configurations.